gcpkms/keysource.go

```diff
@@ -151,3 +153,19 @@ func (key MasterKey) ToMap() map[string]interface{} {
 	out["created_at"] = key.CreationDate.UTC().Format(time.RFC3339)
 	return out
 }
+
+// getDefaultApplicationCredentials allows for passing GCP Service Account
+// Credentials as either a path to a file, or directly as an environment variable
+// in JSON format.
+func getDefaultApplicationCredentials() (token []byte, err error) {
```

Comment:
Why use this when https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentials exists? Shouldn't we check if it returns an error first and then do this custom implementation?

Reply:
It would be worth:

* Checking if `GOOGLE_CREDENTIALS` is set, in which case [`CredentialsFromJSON`](https://pkg.go.dev/golang.org/x/oauth2/google#CredentialsFromJSON) can be called with its content
* Otherwise, call [`FindDefaultCredentials`](https://pkg.go.dev/golang.org/x/oauth2/google#FindDefaultCredentials) and directly return its result

So that we have the special use-case when `GOOGLE_CREDENTIALS` is set, otherwise sops fallback to the default behavior of the SDK, what do you think?

Reply:
I agree, though I'd suggest that we reverse the order. It's good to stay as close to the common cloud provider SDK's as possible and this would ensure we don't break any backwards compatibility.

Reply:
I think @multani's suggestion already is backwards compatible. Given it is not a standard environment variable at the moment, there is some intent for it to be used.

A regular user would not have this set, while by changing it around, it would not be taken into account on GCP instances where the right environment default values are automatically inject at times. Which makes `FindDefaultCredentials` always work, and `GOOGLE_CREDENTIALS` unreachable.

Reply:
I agree with @multani's thoughts here. I'll get the the call to `FindDefaultCredentials` integrated. Seems like it would be better to support both cases `GOOGLE_CREDENTIALS` and `GOOGLE_APPLICATION_CREDENTIALS` to integrate with a wider set of implementations.

Reply:
Ok sounds good.